### PR TITLE
Sonobi Bid Adapter: send eids in bid request.

### PR DESCRIPTION
**Pull request by JonGoSonobi**
## Type of change
- [x] Bugfix

## Description of change
Sending list of eids to sonobi bid request endpoint.
Unwrapping id5id.

- apex.prebid@sonnobi.com
- [x] official adapter submission

### CODE REVIEW

File: test/spec/modules/sonobiBidAdapter_spec.js [hunk 1]
```diff
@@ -414,15 +414,92 @@ describe('SonobiBidAdapter', function () {
       expect(JSON.parse(bidRequests.data.schain)).to.deep.equal(bidRequest[0].schain)
     });
 
+    it('should return a properly formatte drequest with eids as a JSON-encoded set of eids', function () {
```
Comment (smenzer) [inline test/spec/modules/sonobiBidAdapter_spec.js:417]: spelling mistake

```suggestion
    it('should return a properly formatted request with eids as a JSON-encoded set of eids', function () {
```

File: modules/sonobiBidAdapter.js [hunk 2]
```diff
@@ -117,9 +116,21 @@ export const spec = {
       payload.schain = JSON.stringify(validBidRequests[0].schain)
     }
     if (deepAccess(validBidRequests[0], 'userId') && Object.keys(validBidRequests[0].userId).length > 0) {
-      payload.userid = JSON.stringify(validBidRequests[0].userId);
+
+      const userIds = validBidRequests[0].userId;
+      if(userIds.id5id) {
+        userIds.id5id = deepAccess(userIds, 'id5id.uid');
```
Comment (aleksatr) [inline modules/sonobiBidAdapter.js:122]: In general, this is not a good practice, you're modifying the source object here, but in this case, it should be safe as I tested it (and checked the core code) and it appears you're getting an adapter-specific copy of validBidRequests.

**Comment by aleksatr** [comment]
@JonGoSonobi you have lint errors, check circle ci output.

File: modules/sonobiBidAdapter.js (hunk 2 above)
Comment (JonGoSonobi) [reply modules/sonobiBidAdapter.js:124]: Good catch. I copied the userId object so it doesnt get altered when updating id5id here.

**Comment by JonGoSonobi** [comment]
@aleksatr lint errors were fixed bu the CI is still failing on a pubmatic analytics adapter test.

File: modules/sonobiBidAdapter.js (hunk 2 above)
Comment (aleksatr) [reply modules/sonobiBidAdapter.js:124]: are you sure that this way you're actually cloning the object properly? Check this example:
image
There's utils.deepClone() inside pbjs that you can utilize for cloning.

**Comment by aleksatr** [comment]
> @aleksatr lint errors were fixed bu the CI is still failing on a pubmatic analytics adapter test.

np, went through on rerun.

**Comment by aleksatr** [comment]
LGTM

**Pull request merged**